plugins/techdocs/src/Router.tsx

```diff
@@ -69,10 +68,10 @@ export const EmbeddedDocsRouter = (props: PropsWithChildren<{}>) => {
   }
 
   return (
-    <FlatRoutes>
```

Comment:
Hmmmmm.  This breaks Addons on the entity page.

Reply:
Here's a diff you can apply to this branch or master for illustration purposes:

```diff
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 26068c6c8f..a0a5d87c77 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -138,6 +138,8 @@ import {
 import { EntityGoCdContent, isGoCdAvailable } from '@backstage/plugin-gocd';
 
 import React, { ReactNode, useMemo, useState } from 'react';
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { TextSize } from '../techdocs/TextSize';
 
 const customEntityFilterKind = ['Component', 'API', 'System'];
 
@@ -397,7 +399,11 @@ const serviceEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
 
     <EntityLayout.Route
@@ -464,7 +470,11 @@ const websiteEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
     <EntityLayout.Route
       if={isNewRelicDashboardAvailable}
@@ -503,7 +513,11 @@ const defaultEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/todos" title="TODOs">
diff --git a/packages/app/src/components/techdocs/TextSize.tsx b/packages/app/src/components/techdocs/TextSize.tsx
new file mode 100644
index 0000000000..9c410dd052
--- /dev/null
+++ b/packages/app/src/components/techdocs/TextSize.tsx
@@ -0,0 +1,51 @@
+import React, { useEffect, useCallback, useState } from 'react';
+import { Tooltip, ButtonGroup, Button } from '@material-ui/core';
+import ZoomInIcon from '@material-ui/icons/ZoomIn';
+import ZoomOutIcon from '@material-ui/icons/ZoomOut';
+
+import { createTechDocsAddonExtension, TechDocsAddonLocations, useShadowRoot } from '@backstage/plugin-techdocs-react';
+import { techdocsPlugin } from '@backstage/plugin-techdocs';
+
+const TYPESET = '.md-typeset';
+
+export const TextSizeAddon = () => {
+  const shadowRoot = useShadowRoot();
+  const [fontSize, setFontSize] = useState<number>(16);
+
+  useEffect(() => {
+    const typeset = shadowRoot?.querySelector<HTMLElement>(TYPESET);
+    if (!typeset) return;
+    typeset.style.fontSize = `${fontSize}px`;
+  }, [shadowRoot, fontSize]);
+
+  const handleZoomIn = useCallback(() => {
+    setFontSize(fontSize + 1);
+  }, [fontSize, setFontSize]);
+
+  const handleZoomOut = useCallback(() => {
+    setFontSize(fontSize - 1);
+  }, [fontSize, setFontSize]);
+
+  return (
+    <ButtonGroup variant="outlined">
+      <Tooltip title="Text Zoom In" arrow>
+        <Button size="small" onClick={handleZoomIn} aria-label="text zoom in">
+          <ZoomInIcon />
+        </Button>
+      </Tooltip>
+      <Tooltip title="Text Zoom Out" arrow>
+        <Button size="small" onClick={handleZoomOut} aria-label="text zoom out">
+          <ZoomOutIcon />
+        </Button>
+      </Tooltip>
+    </ButtonGroup>
+  );
+};
+
+export const TextSize = techdocsPlugin.provide(
+  createTechDocsAddonExtension({
+    name: 'TextSize',
+    location: TechDocsAddonLocations.Subheader,
+    component: TextSizeAddon,
+  }),
+);

```

Reply:
Indeed, fixed! There was a bit more wrapping than the element discovery was able to handle. Consider switching to [useElementFilter](https://backstage.io/docs/reference/core-plugin-api.useelementfilter) though. I see there's a bit much custom code when traversing the elements, which makes it a bit fragile.

Reply:
Indeed.